hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php

```diff
@@ -145,42 +131,40 @@ function testSyncSncenario2() {
     $this->assertEquals(CRM_Core_OptionGroup::getValue('activity_status', 'Cancelled', 'name'), $activity['values'][$activity['id']]['status_id'], 'in line ' . __LINE__ . ' Status of \'Visa Expiration\' activity should be \'Cancelled\' but wrongly is ' . $activity['values'][$activity['id']]['status_id']);
   }
 
-  function _getTargetContactActivity($contactId) {
+  private function getTargetContactActivity($contactId) {
     $activityTypeId = CRM_Core_OptionGroup::getValue('activity_type', 'Visa Expiration', 'name');
     // to check if visa expiration activity exists for the input target_contact_id
     $activityGetParams = array(
       'contact_id' => $contactId,
       'activity_type_id' => $activityTypeId,
       'sequential' => 1,
     );
-    // note : using filter 'activity_type_id' in combination with 'contact_id' filter doesn't work
     $activities = civicrm_api3('activity', 'get', $activityGetParams);
 
     $activityId = NULL;
     $count = 0;
     foreach($activities['values'] as $val) {
-      if ($val['activity_type_id'] != $activityTypeId || !array_key_exists('targets', $val)) {
-        continue;
-      }
       $activityId = $val['id'];
       $count++;
     }
+
     return array($count, $activityId);
   }
 
-  /**
-   * Helper function to load data into DB between iterations of the unit-test
-   */
-  private static function phpunitPopulateDB() {
-    $import = new CRM_Utils_Migrate_Import();
-    $import->run(
-      CRM_Extension_System::singleton()->getMapper()->keyToBasePath('org.civicrm.hrvisa')
-      . '/xml/auto_install.xml'
-    );
-    // this had to be done as demographics consists of is_visa_required field (used in unit test)
-    $import->run(
-      CRM_Extension_System::singleton()->getMapper()->keyToBasePath('org.civicrm.hrdemog')
-      . '/xml/auto_install.xml'
-    );
+  private function createLoggedInUser() {
```

Comment:
That's a great a idea. In fact, in L&A we do have a trait that does almost the same thing: https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/helpers/SessionHelpersTrait.php

I even thought about adding that to this test, but then I found a few reasons to not do it (for now):
- That trait it's on the 1.7 branch, so I would have to copy it to the 1.6 branch in order to be able to use it here
- It works on a slightly different way than the one we have in SessionHelpersTrait
- I could create a new trait just for this test, but then I would have 2 different traits doing almost the same thing, which is not a good thing

So, my idea is to wait for this until we have 1.7 merged into staging. Also, once we have Jenkins, we'll have to fix and improve a lot of things on the existing old tests. One of these things we'll definitely be extracting this to a generic trait that can be used both by hrvisa and L&A